crates/rattler_installs_packages/src/python_env/venv.rs

```diff
@@ -291,11 +290,32 @@ prompt = {}"#,
                 .expect("cannot get system python parent folder");
             for bin_name in python_bins.into_iter() {
                 let original_python_bin = original_python_bin_dir.join(bin_name);
+                let original_python_scripts = original_python_bin_dir
+                    .join("Lib/venv/scripts/nt")
+                    .join(bin_name);
+                let venv_python_bin = venv_bin.join(bin_name);
 
-                if original_python_bin.exists() {
-                    let venv_python_bin = venv_bin.join(bin_name);
-                    if !venv_python_bin.exists() {
-                        copy_file(venv_exe_path, &venv_python_bin)?;
+                if original_python_bin.exists() && !venv_python_bin.exists() {
+                    if original_python_scripts.is_file() {
+                        copy_file(original_python_scripts, &venv_python_bin)?;
+                    } else {
+                        // If used python is built from source code
+                        // or we are using a python from venv which
+                        // was created using python from source code
+                        // we need to move venvlauncher
+                        // and venvwlauncher as python.exe and pythonw.exe
+                        // instead of using the python.exe shipped with /venv/scripts/nt
+                        let launcher_bin_name = if bin_name.contains("python") {
+                            bin_name.replace("python", "venvlauncher")
+                        } else if bin_name.contains("pythonw") {
+                            bin_name.replace("pythonw", "venvwlauncher")
+                        } else {
+                            bin_name.to_owned()
+                        };
+                        let original_launcher_bin = original_python_bin_dir.join(launcher_bin_name);
+                        if original_launcher_bin.exists() {
+                            copy_file(original_launcher_bin, &venv_python_bin)?;
+                        }
```

Comment:
Do we ever hit this case? According to the original venv module this only happens when symlinking on windows and using a python that is built from source at the location it was built, so not even when it is moved by a binary install.

Reply:
No, in our cases we don't ( I could not reproduce it ) but I've decided to keep it in case our users will pass a built python as location